digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/ProjectDAO.scala

```diff
@@ -439,28 +444,82 @@ object ProjectDAO {
       ""
     val sql =
       s"""
-           SELECT id, road_number, road_part_number, road_length, ADDRESS_LENGTH, discontinuity, ely, first_link_id,
-           (select PROJECT_LINK.ID from PROJECT_LINK WHERE PROJECT_LINK.ROAD_NUMBER = rp.ROAD_NUMBER AND
-           PROJECT_LINK.ROAD_PART_NUMBER = rp.ROAD_PART_NUMBER AND
-           PROJECT_LINK.PROJECT_ID = rp.PROJECT_ID AND ROWNUM < 2) as pl_id
-           FROM PROJECT_RESERVED_ROAD_PART rp WHERE project_id = $projectId $filter ORDER BY road_number, road_part_number
-         """
-    Q.queryNA[(Long, Long, Long, Double, Long, Int, Long, Option[Long], Option[Long])](sql).list.map {
-      case (id, road, part, length, addrLength, discontinuity, ely, linkId, plId) => ReservedRoadPart(id, road, part,
-        length, addrLength, Discontinuity.apply(discontinuity), ely, None, None, linkId, plId.nonEmpty)
+        SELECT id, road_number, road_part_number, length, length_new,
+          ely, ely_new,
+          (SELECT DISCONTINUITY FROM ROAD_ADDRESS ra WHERE ra.road_number = gr.road_number AND
+          ra.road_part_number = gr.road_part_number AND RA.END_DATE IS NULL AND RA.VALID_TO IS NULL
+          AND END_ADDR_M = gr.length and ROWNUM < 2) as discontinuity,
+          (SELECT DISCONTINUITY_TYPE FROM PROJECT_LINK pl WHERE pl.project_id = gr.project_id
+          AND pl.road_number = gr.road_number AND pl.road_part_number = gr.road_part_number
+          AND PL.STATUS != 5 AND PL.TRACK_CODE IN (0,1)
+          AND END_ADDR_M = gr.length_new) as discontinuity_new,
+          (SELECT LINK_ID FROM PROJECT_LINK pl JOIN LRM_POSITION lrm ON (lrm.id = pl.LRM_POSITION_ID)
+            WHERE pl.project_id = gr.project_id
+            AND pl.road_number = gr.road_number AND pl.road_part_number = gr.road_part_number
+            AND PL.STATUS != 5 AND PL.TRACK_CODE IN (0,1) AND pl.START_ADDR_M = 0
+            AND pl.END_ADDR_M > 0 AND ROWNUM < 2) as first_link
+          FROM (
+            SELECT rp.id, rp.project_id, rp.road_number, rp.road_part_number,
+              MAX(ra.END_ADDR_M) as length,
+              MAX(pl.END_ADDR_M) as length_new,
+              MAX(ra.ely) as ELY,
+              MAX(pl.ely) as ELY_NEW
+              FROM PROJECT_RESERVED_ROAD_PART rp LEFT JOIN
+              ROAD_ADDRESS ra ON (ra.road_number = rp.road_number AND ra.road_part_number = rp.road_part_number)
+              LEFT JOIN
+              PROJECT_LINK pl ON (pl.project_id = rp.project_id AND pl.road_number = rp.road_number AND pl.road_part_number = rp.road_part_number)
+              WHERE
+                rp.project_id = $projectId AND
+                RA.END_DATE IS NULL AND RA.VALID_TO IS NULL AND
+                (PL.STATUS IS NULL OR (PL.STATUS != 5 AND PL.TRACK_CODE IN (0,1)))
```

Comment:
Why are we excluding LeftSide tracks?

Reply:
Well i believe its because the we are getting the road part length with this query and since the endaddress from last track 1 should be the same has endaress from last track2 getting one of them is enough to get the road part length